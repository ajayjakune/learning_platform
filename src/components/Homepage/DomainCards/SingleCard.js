import React, {useState} from 'react'
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import './Card.css'


const SingleCard = (props) => {
    // console.log(props.Title)
    // console.log(props.Description)


    
    return (
        <>
            {/* <div className="col-lg-4 col-md-6 col-xs-12" style={{ marginBottom: "30px"}}>
                <div className="card">
                    <div className="card-img-wrapper">
                        <img className="card-img-top" src={props.ImageUrl} alt="Card image cap"></img>
                    </div>
                    <div className="card-body">
                        <h5 className="card-title">{props.Title}</h5>
                        <p className="card-text">{props.Description}</p>
                        <a href="#" className="btn btn-primary">{props.button}</a>
                    </div>
                </div>
            </div> */}

            <div className="card-wrapper col-lg-4 col-md-6 col-xs-12">
            <Card style={{ marginTop: "10px", marginBottom: "20px", height: "300px", borderRadius: "15px", }} className="fcard" > 
            <Card.Img variant="top" src={props.ImageUrl} style={{ objectFit: "cover", height: "180px", borderTopRightRadius: "15px", borderTopLeftRadius: "15px"}} />
            <Card.Body>
                <Card.Title>{props.Title}</Card.Title> 
                {/* <Card.Text>
                    {props.Description}
                </Card.Text> */}
                <Button variant="primary" className="fbtn" onClick={() => props.setselectedCard(props.Title)} >{props.button}</Button>
            </Card.Body>
            </Card>
            </div>
        </>
    )
}

export default SingleCard
