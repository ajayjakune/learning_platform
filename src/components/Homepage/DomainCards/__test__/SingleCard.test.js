import React from 'react'
import {BrowserRouter} from 'react-router-dom'
import SingleCard from '../SingleCard'
import renderer from 'react-test-renderer'


// it("Renders without crashing", () => {
//     const div = document.createElement("div")
//     ReactDOM.render(<SingleCard />, div)
//     ReactDOM.unmountComponentAtNode(div)
// })

// it("Renders without crashing", () =>  { 
//     const div = document.createElement("div")
//     ReactDOM.render(<SingleCard />, div)
//     ReactDOM.unmountComponentAtNode(div)
// })

test("SingleCard matches the snapshot", () => {
    const imageUrl = "https://www.google.com"
    const title = "AI"
    const description = "Its some tech"
    const cardId = "1"
    const buttonText = "Explore !" 
    const tree = renderer.create(
        <>
        <BrowserRouter>
        <SingleCard imageUrl= {imageUrl} cardId={cardId} title={title} description={description} buttonText={buttonText} />
        </BrowserRouter>
        </>).toJSON()

    console.log(tree)
    expect(tree).toMatchSnapshot()

})