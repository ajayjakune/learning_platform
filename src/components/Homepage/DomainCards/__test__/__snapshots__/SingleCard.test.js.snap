// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCard matches the snapshot 1`] = `
<div
  className="card-wrapper cus-card-wrapper col-lg-4 col-md-6 col-xs-12"
  style={
    Object {
      "marginBottom": "30px",
    }
  }
>
  <div
    className="card cus-card"
  >
    <div
      className="card-img-wrapper cus-card-img-wrapper"
    >
      <img
        alt="Card image cap"
        className="card-img-top cus-card-img-top"
        src="https://www.google.com"
      />
    </div>
    <div
      className="card-body cus-card-body"
    >
      <h5
        className="card-title cus-card-title"
      >
        AI
      </h5>
      <p
        className="card-text cus-card-text"
      >
        Its some tech
      </p>
      <a
        href="/course/1"
        onClick={[Function]}
      >
        <div
          className="btn btn-primary fbtn"
        >
          Explore !
        </div>
      </a>
    </div>
  </div>
</div>
`;
